Src/Support/GoogleApis/Apis/Util/Store/FileDataStore.cs

```diff
@@ -0,0 +1,179 @@
+﻿/*
+Copyright 2017 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#if NETSTANDARD
+
+// Only for NetStandard as Desktop already has a FileDataStore
+// in the PlatformServices dll.
+// This is slightly unforunate, but cannot now easily be changed.
+// The FileDataStore here is identical, except for platform-specific
+// selection of the application data directory.
+
+// TODO: This does not support UWP Storage.
+
+using Google.Apis.Json;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Google.Apis.Util.Store
+{
+    /// <summary>
+    /// File data store that implements <see cref="IDataStore"/>. This store creates a different file for each 
+    /// combination of type and key. This file data store stores a JSON format of the specified object.
+    /// </summary>
+    public class FileDataStore : IDataStore
+    {
+        readonly string folderPath;
+        /// <summary>Gets the full folder path.</summary>
+        public string FolderPath { get { return folderPath; } }
+
+        /// <summary>
+        /// Constructs a new file data store. If <c>fullPath</c> is <c>false</c> the path will be used as relative to 
+        /// <see cref="Environment.SpecialFolder.ApplicationData"/> on Windows, or $HOME on Linux and MacOS,
+        /// otherwise the input folder will be treated as absolute.
+        /// The folder is created if it doesn't exist yet.
+        /// </summary>
+        /// <param name="folder">Folder path.</param>
+        /// <param name="fullPath">
+        /// Defines weather the folder parameter is absolute or relative to
+        /// <see cref="Environment.SpecialFolder.ApplicationData"/>.
+        /// </param>
+        public FileDataStore(string folder, bool fullPath = false)
+        {
+            folderPath = fullPath
+                ? folder
+                : Path.Combine(GetHomeDirectory(), folder);
+            if (!Directory.Exists(folderPath))
+            {
+                Directory.CreateDirectory(folderPath);
+            }
+        }
+
+        private string GetHomeDirectory()
+        {
+            string path =
+                Environment.GetEnvironmentVariable("APPDATA") ?? // Windows
+                Environment.GetEnvironmentVariable("HOME"); // Linux, MacOS
+            if (path == "")
```

Comment:
You're assuming that `Environment.GetEnvironmentVariable("APPDATA")` will return `null` if it's missing a couple of lines above, but now we're assuming `Environment.GetEnvironmentVariable("HOME")` will return `""`?